### PR TITLE
Implement DELETE and PATCH method tests for support-notification subscription

### LINKED ISSUE #343: Implement tests for support-notification subscription
Implement tests for support-notification subscription methods.
1. GET
2. DELETE
3. PATCH

### PR DESCRIPTION
**Pull request by cherrycl**
Implement DELETE and PATCH method tests for support-notification subscription
Update channel_data for DTO changing.
Fix #343 
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

### CODE REVIEW

**Comment by cherrycl** [comment]
Raised [edgex-go issue#3298](https://github.com/edgexfoundry/edgex-go/issues/3298) for failure tests.

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Negative.robot [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    When Update Subscriptions ${subscription}
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Subscriptions By Names  @{subscription_names}
+
+ErrSubscriptionPATCH007 - Update subscription for REST channel with empty Host
+    Given Generate A Subscription Sample With REST Channel
+    And Create Subscription ${subscription}
+    And Set To Dictionary  ${subscription}[0][subscription][channels][0]  Host=${EMPTY}
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Negative.robot:80]:
```suggestion
    And Set To Dictionary  ${subscription}[0][subscription][channels][0]  host=${EMPTY}
```

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Positive.robot [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    ${update_channels}  Create Dictionary  name=${subscription_names}[2]  channels=${channels}  description=update subscription
+    ${update_receiver}  Create Dictionary  name=${subscription_names}[3]  receiver=updateuser
+    Generate Subscriptions  ${update_labels}  ${update_categories}  ${update_channels}  ${update_receiver}
+
+Subscriptions Should Be Updated
+    ${port}  Convert To Integer  8888
+    ${limit}  Convert To Integer  3
+    FOR  ${name}  IN  @{subscription_names}
+        Query Subscription By Name ${name}
+        ${keys}=  Get Dictionary Keys  ${content}[subscription]
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Positive.robot:46]: The variable `keys` is unused.

File: TAF/testCaseModules/keywords/support-notifications/subscriptionAPI.robot [hunk 3]
```diff
@@ -51,18 +51,16 @@ Create Subscription ${entity}
...
+    Set Response to Test Variables  ${resp}
+    Run keyword if  ${response} != 207  log to console  ${content}
 
 Delete Subscription By Name ${subscriptionName}
     Create Session  Support Notification  url=${supportNotificationsUrl}  disable_warnings=true
     ${headers}=  Create Dictionary  Authorization=Bearer ${jwt_token}
     ${resp}=  DELETE On Session  Support Notification  ${subscriptionUri}/name/${subscriptionName}  headers=${headers}
     ...       expected_status=any
     run keyword if  ${resp.status_code}!=204  log to console  ${resp.content}
+    Set Suite Variable  ${response}  ${resp.status_code}
```
Comment (jinlinGuan) [inline TAF/testCaseModules/keywords/support-notifications/subscriptionAPI.robot:63]:
```suggestion
    Set Response to Test Variables ${resp}
```
This keyword returns `${response}`, `${headers}`, `${response_time}` and `${content}`

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Negative.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Negative.robot]: Fix.

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Positive.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/PATCH-Positive.robot]: Fix.

File: TAF/testCaseModules/keywords/support-notifications/subscriptionAPI.robot (hunk 3 above)
Comment (cherrycl) [reply TAF/testCaseModules/keywords/support-notifications/subscriptionAPI.robot:64]: Use `set suite variable` temporarily after the [edgex-go issue #3308](https://github.com/edgexfoundry/edgex-go/issues/3308) is fixed.

Comment (cherrycl) [reply TAF/testCaseModules/keywords/support-notifications/subscriptionAPI.robot]: Fix. Opened [ edgex-go issue #3308 ](https://github.com/edgexfoundry/edgex-go/issues/3308) for delete API error.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**